src/Factory/Form/MainSettingsFormTypeFactory.php

```diff
@@ -68,12 +68,21 @@ private function getInitialFormData(SettingsInterface $settings): array
         return $data + $this->getChannelInitialFormData($settings);
     }
 
+    /**
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
     private function getChannelInitialFormData(SettingsInterface $settings): array
     {
         $data = [];
         /** @var ChannelInterface $channel */
         foreach ($this->channelRepository->findAll() as $channel) {
-            $data['channel-' . $channel->getId() . '-' . Settings::DEFAULT_KEY] = $settings->getSettingsValuesByChannelAndLocale($channel);
+            $channelKey = 'channel-' . $channel->getId() . '-' . Settings::DEFAULT_KEY;
+            $data[$channelKey] = $settings->getSettingsValuesByChannelAndLocale($channel);
+            $defaultDataByChannel = $this->getDefaultValuesByChannel($settings, $channel);
+
+            if (isset($defaultDataByChannel[$channelKey])) {
+                $data[$channelKey] += $defaultDataByChannel[$channelKey];
+            }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏why the defaults value per channel are added here but not the global defaults?

Reply:
because we are in getChannelInitialFormData() the global default value is handle before the call to this method